allennlp/commands/fine_tune.py

```diff
@@ -188,14 +188,15 @@ def fine_tune_model(model: Model,
     for name, parameter in model.named_parameters():
         if any(re.search(regex, name) for regex in no_grad_regexes):
             parameter.requires_grad_(False)
-            nograd_parameter_names.append(name)
-        else:
+        if parameter.requires_grad:
             grad_parameter_names.append(name)
+        else:
+            nograd_parameter_names.append(name)
 
-    logger.info("Following parameters are Frozen  (without gradient):")
+    logger.info("FROZEN PARAMETERS  (WITHOUT GRADIENT):")
     for name in nograd_parameter_names:
         logger.info(name)
-    logger.info("Following parameters are Tunable (with gradient):")
```

Comment:
Perhaps consider adding a small test for this - given the same code is in both places, consider pulling it out into a function, which would make testing it easier.

Reply:
Sure. 

Do you mean I add `def get_frozen_and_tunable_parameter_names(parameters: List[Parameter]) -> List` in different place like `allennlp/common/util.py` and use it in both `train.py` and `fine_tune.py`? Something like following?

```python
# allennlp/commons/util.py
def get_frozen_and_tunable_parameter_names(model: torch.nn.Module) -> List:
    frozen_parameter_names = []
    tunable_parameter_names = []
    for name, parameter in model.named_parameters():
        if not parameter.requires_grad:
            frozen_parameter_names.append(name)
        else:
            tunable_parameter_names.append(name)
    return [frozen_parameter_names, tunable_parameter_names]
```
and test could be something like:
```python
def test_get_frozen_and_tunable_parameter_names(self):
    model = torch.nn.Sequential(OrderedDict([
          ('conv', torch.nn.Conv1d(5, 5, 5)),
          ('linear', torch.nn.Linear(5, 10)),
        ])) 
    named_parameters = dict(model.named_parameters())
    named_parameters['linear.weight'].requires_grad_(False)
    named_parameters['linear.bias'].requires_grad_(False)
    frozen_parameter_names, tunable_parameter_names = \
                   util.get_frozen_and_tunable_parameter_names(model)
    assert set(frozen_parameter_names)  == {'linear.weight', 'linear.bias'}
    assert set(tunable_parameter_names) == {'conv.weight', 'conv.bias'}
```
and train and fine-tune would be:
```python

    frozen_parameter_names, tunable_parameter_names = \
                   util.get_frozen_and_tunable_parameter_names(model)
    logger.info("Following parameters are Frozen  (without gradient):")
    for name in frozen_parameter_names:
        logger.info(name)
    logger.info("Following parameters are Tunable (with gradient):")
    for name in tunable_parameter_names:
        logger.info(name)
```

Seems good?

Reply:
Perfect 👍